pinot-spi/src/main/java/org/apache/pinot/spi/stream/MessageBatch.java

```diff
@@ -116,4 +116,14 @@ default StreamPartitionMsgOffset getOffsetOfNextBatch() {
   default boolean isEndOfPartitionGroup() {
     return false;
   }
+
+  /**
+   * This is useful while determining ingestion delay for a message batch. Retaining metadata for last message in
+   * a batch can enable us to estimate the ingestion delay for the batch.
```

Comment:
Can you also add an example on how the batch may look like?
Also mention that it is possible for the batch to be empty (all data is filtered), in which case the row metadata may still be non-null to facilitate ingestion delay computation.

Reply:
Done